designs/tinkerbell-provider-input.md

```diff
@@ -0,0 +1,366 @@
+```
+
+For Tinkerbell provider, we have introduced two new CRDs:
+* TinkerbellDatacenterConfig
+* TinkerbellMachineConfig
+
+**TinkerbellDatacenterConfig**
+
+TinkerbellDatacenterConfig describes the configuration details for Tinkerbell stack.
+
+In this config, users must specify the following fields:
+* spec.tinkerbellIP: defines the IP of the machine running the Tinkerbell stack.
+* spec.tinkerbellCertURL: defines the endpoint where the Tinkerbell stack hosts its certificate. This is used to communicate with the stack.
+* spec.tinkerbellGRPCAuth: defines the GRPC endpoint used by Tinkerbell stack. This is also used to communicate with the stack.
+* spec.tinkerbellPBnJGRPCAuth: defines the GRPC endpoint used by PNBJ for hardware remote management. This is used to change the power state and the boot order of the servers.
+
+Here's an example `TinkerbellDatacenterConfig`
+
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: TinkerbellDatacenterConfig
+metadata:
+  name: eks-a-tinkerbell-cluster
+spec:
+  tinkerbellIP: "192.168.0.20"
+  tinkerbellCertURL: "http://192.168.0.20:42114/cert"
+  tinkerbellGRPCAuth: "192.168.0.20:42113"
+  tinkerbellPBnJGRPCAuth: "192.168.0.20:50051"
+```
+
+**TinkerbellMachineConfig**
+
+TinkerbellMachineConfig describes the configuration for machines used for creating EKS Anywhere cluster on Tinkerbell.
+
+This config supports the following fields
+* spec.osFamily (required): defines the operating system to be installed on this machine. We currently only support ubuntu for this but this list is expected to grow in the future.
+* spec.users (required): defines the users to be created on this machine along with the ssh keys.
+  * spec.users[].name (required): defines the name of the user.
+  * spec.users[].sshAuthorizedKeys (required): defines the SSH authorized keys to be installed on this machine to be used for SSH'ing into it.
+* spec.templateOverride (optional): defines the [Tinkerbell template](https://docs.tinkerbell.org/templates/) to be used to provision the OS on the machine. This field is passed as a YAML encoded as string.
+
+Here is an example `TinkerbellMachineConfig`
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: TinkerbellMachineConfig
+metadata:
+  name: eks-a-tinkerbell-cluster-cp
+spec:
+  osFamily: ubuntu
+  users:
+  - name: ec2-user
+    sshAuthorizedKeys:
+    - ssh-rsa AAAA...
+  templateOverride: |
+    <>
+```
+
+Here is the default `templateOverride` that is generated in the clusterconfig
+```yaml
+version: "0.1"
+name: eks-a-tinkerbell-cluster-cp-test
+global_timeout: 6000
+tasks:
+  - name: "eks-a-tinkerbell-cluster-cp-ndrl8"
+    worker: "{{.device_1}}"
+    volumes:
+      - /dev:/dev
+      - /dev/console:/dev/console
+      - /lib/firmware:/lib/firmware:ro
+    actions:
+      - name: "stream-image"
+        image: image2disk:v1.0.0
+        timeout: 360
+        environment:
+          IMG_URL: "https://<some-s3-endpoint>/ubuntu-2004-kube-v1.21.5.gz"
```

Comment:
Doesn't have to be s3 right?

Reply:
right can be any endpoint. I said s3 because that's where we host our images